File: monotonic_array.js
const checkMonotonic = function(array) {
    //write code here to return either true or false 
    let a = parseInt(array[0]);
    let b = parseInt(array[1]);

    if ((array.length == 0) || (array.length == 1)) {
        return true;
    }

    let increasing = false;
    let decreasing = false;

    if (a <= b) {
        increasing = true;
    } else {
        decreasing = true;
    }

    for (let i = 2; i < array.length; i++) {
        if (increasing == true) {
            if (array[i - 1] > array[i]) {
                return false;
            }
        }
        if (decreasing == true) {
            if (array[i - 1] < array[i]) {
                return false;
            }
        }
    }

    return true;
}

File: monotonic_array.py
def monotonic_array(array):
    #write code here
    if len(array)==1 or len(array)==0:
        return True
    i=array[0]
    j=array[1]
    
    increasing = False
    decreasing = False
    
    if i>=j:
        decreasing = True
    else:
        increasing = True
    
        
    for i in range(2,len(array)):
        if increasing:
            if array[i-1]>=array[i]:
                return False
        
        if decreasing:
            if array[i-1]<=array[i]:
                return False
    
    return True